### PR TITLE
sapl-spring-data: fixed pull request comments and sonar issues

### PR DESCRIPTION
**Pull request by RooSchoo**
## Checklist

* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.
* [ ] All new and existing tests passed.
* [ ] My code follows the code style of this project.
* [x] I ran all checks which produced no new errors or warnings for my changes.
* [ ] I have checked to ensure there aren't other open Pull Requests for the same update/change.

### CODE REVIEW

File: sapl-spring-data-mongo-reactive/src/main/java/io/sapl/springdatamongoreactive/sapl/queries/enforcement/MongoAnnotationQueryManipulationEnforcementPoint.java [hunk 1]
```diff
@@ -48,7 +49,7 @@
  * @param <T> is the domain type.
  */
 public class MongoAnnotationQueryManipulationEnforcementPoint<T> implements QueryManipulationEnforcementPoint<T> {
-    private static final String MONGO_QUERY_MANIPULATION = "mongoQueryManipulation";
+    private static final String mongoQueryManipulation = "mongoQueryManipulation";
```
Comment (heutelbeck) [inline sapl-spring-data-mongo-reactive/src/main/java/io/sapl/springdatamongoreactive/sapl/queries/enforcement/MongoAnnotationQueryManipulationEnforcementPoint.java:52]: you still overwrite the correct case

File: sapl-spring-data-mongo-reactive/src/main/java/io/sapl/springdatamongoreactive/sapl/queries/enforcement/ReflectedMongoQueryCreator.java [hunk 2]
```diff
@@ -32,7 +32,8 @@ protected ReflectedMongoQueryCreator() {
         this.mongoQueryCreator = Class.forName(MONGO_QUERY_CREATOR_NAME);
     }
 
-    protected Constructor<?> getDeclaredConstructor(Class<?> class1, Class<?> class2, Class<?> class3, Class<?> class4)
+    @SuppressWarnings("rawtypes")
```
Comment (heutelbeck) [inline sapl-spring-data-mongo-reactive/src/main/java/io/sapl/springdatamongoreactive/sapl/queries/enforcement/ReflectedMongoQueryCreator.java:35]: @SupressWarnings still not justified in general

**Review by heutelbeck — CHANGES_REQUESTED** [review]
Please also address the issues from SonarCloud and SonarQube.

File: sapl-spring-data-mongo-reactive/src/main/java/io/sapl/springdatamongoreactive/sapl/queries/enforcement/ReflectedMongoQueryCreator.java (hunk 2 above)
Comment (RooSchoo) [reply sapl-spring-data-mongo-reactive/src/main/java/io/sapl/springdatamongoreactive/sapl/queries/enforcement/ReflectedMongoQueryCreator.java:35]: Yes I know, this should not be the last PR

Comment (RooSchoo) [reply sapl-spring-data-mongo-reactive/src/main/java/io/sapl/springdatamongoreactive/sapl/queries/enforcement/ReflectedMongoQueryCreator.java:35]: How do I justify a SupressWarning? What is meant by justifying a warning?

**Comment by heutelbeck** [comment]
Fix R2dbcEntityTemplateExecutorTests there is only commented out code.

**Comment by RooSchoo** [comment]
> Fix R2dbcEntityTemplateExecutorTests there is only commented out code.

I would love to fix it and i already spent few hours on this one but when mocking R2dbcEntityTemplate I always get this error: 

Simple Test: 

    private final static String QUERY = "SELECT * FROM person WHERE firstname = 'Melinda'";

    @Test
    void when_executeQuery_then_callR2dbcEntityTemplateFetch() {
        // GIVEN
        var r2dbcEntityTemplateMock     = mock(R2dbcEntityTemplate.class, Answers.RETURNS_DEEP_STUBS);
        var r2dbcEntityTemplateExecutor = new R2dbcEntityTemplateExecutor(r2dbcEntityTemplateMock);

        // WHEN
        r2dbcEntityTemplateExecutor.executeQuery(QUERY);

        // THEN
        verify(r2dbcEntityTemplateMock.getDatabaseClient(), times(1)).sql(QUERY);
    }

Error: 

java.lang.NullPointerException: Cannot invoke "[Ljava.lang.Class;.clone()" because "<local2>.parameterTypes" is null
	at java.base/java.lang.reflect.Method.getParameterTypes(Method.java:314)
	at org.mockito.internal.creation.DelegatingMethod.<init>(DelegatingMethod.java:20)
	at org.mockito.internal.invocation.DefaultInvocationFactory.createMockitoMethod(DefaultInvocationFactory.java:81)
	at org.mockito.internal.invocation.DefaultInvocationFactory.createInvocation(DefaultInvocationFactory.java:60)
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:83)
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:56)
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptAbstract(MockMethodInterceptor.java:161)
	at org.springframework.r2dbc.core.DatabaseClient$MockitoMock$yWyHDFpH.sql(Unknown Source)
	at io.sapl.springdatar2dbc.sapl.R2dbcEntityTemplateExecutor.executeQuery(R2dbcEntityTemplateExecutor.java:35)
	at io.sapl.springdatar2dbc.sapl.R2dbcEntityTemplateExecutorTest.when_executeQuery_then_callR2dbcEntityTemplateFetch(R2dbcEntityTemplateExecutorTest.java:22)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)

**Comment by heutelbeck** [comment]
maybe use the real class and use a spy instead of a mock ?

**Comment by RooSchoo** [comment]
> maybe use the real class and use a spy instead of a mock ?

good point, but unfortunately I can't do it, because to get a real R2dbcEntityTemplate, I would have to establish a real database connection, which does not exist in this module. The two modules use the database connection of the application in which they are integrated

**Comment by heutelbeck** [comment]
This is a pretty brittle way to test it. Please double-check if you can apply the following style of tests in other places to make them more robust and narrow down the unit under test.

Solution:

```java
class R2dbcEntityTemplateExecutorTests {

    private final static String QUERY = "SELECT * FROM person WHERE firstname = 'Melinda'";

    @Test
    @SuppressWarnings("unchecked") // mocking of generic type
    void when_executeQuery_then_callR2dbcEntityTemplateFetch() {
        // GIVEN
        var dbClientMock            = mock(DatabaseClient.class);
        var fetchSpecMock           = mock(FetchSpec.class);
        var genericExecuteSpecMock  = mock(GenericExecuteSpec.class);
        var r2dbcEntityTemplateMock = mock(R2dbcEntityTemplate.class);

        when(r2dbcEntityTemplateMock.getDatabaseClient()).thenReturn(dbClientMock);
        when(dbClientMock.sql(any(String.class))).thenReturn(genericExecuteSpecMock);
        when(genericExecuteSpecMock.fetch()).thenReturn(fetchSpecMock);

        var r2dbcEntityTemplateExecutor = new R2dbcEntityTemplateExecutor(r2dbcEntityTemplateMock);

        // WHEN
        r2dbcEntityTemplateExecutor.executeQuery(QUERY);

        // THEN
        verify(r2dbcEntityTemplateMock, times(1)).getDatabaseClient();
        verify(dbClientMock, times(1)).sql(QUERY);
        verify(genericExecuteSpecMock, times(1)).fetch();
        verify(fetchSpecMock, times(1)).all();
    }
}
```

**Comment by heutelbeck** [comment]
This narrows down the same error during ```mvn install``` to:

```java
        when(dbClientMock.sql(any(String.class))).thenReturn(genericExecuteSpecMock);
```

**Comment by heutelbeck** [comment]
See: https://github.com/mockito/mockito/issues/2026#issuecomment-740545855

So apparently, there are some weird side-effects from the other tests causing this.

Adding this to the module's POM solves the issue. 
```xml
	<build>
		<plugins>
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-surefire-plugin</artifactId>
				<configuration>
					<forkCount>1</forkCount>
					<reuseForks>false</reuseForks>
				</configuration>
			</plugin>
		</plugins>
	</build>
```

```java
        when(dbClientMock.sql(anyString())).thenReturn(genericExecuteSpecMock);
```

This is nicer than ```any(String.class)```.

**Comment by RooSchoo** [comment]
Yay, it works smoothly now. Thanks for your efforts. 

Did I understand correctly that I rewrite every test that contains a mock with an Answers.RETURNS_DEEP_STUBS?

**Comment by heutelbeck** [comment]
> o

You don't need to do that. If you think they benefit, please do so. I would check if one could pin down the test, causing the side effects that cause the other test to fail. A better solution to fix that test (which could be any other test in the test suite) than disabling forking. The problem is likely not the single test or the forking itself but some other test that misbehaving.

**Comment by heutelbeck** [comment]
Please revolve conflicts. Thank you. Let's get this done.

**Pull request closed without merging**